src/graphics/graphics-device.js

```diff
@@ -3496,8 +3497,8 @@ class GraphicsDevice extends EventHandler {
         this._height = height;
 
         var ratio = Math.min(this._maxPixelRatio, window.devicePixelRatio);
-        width *= ratio;
-        height *= ratio;
+        width = Math.floor(width * ratio);
+        height = Math.floor(height * ratio);
 
         if (this.canvas.width === width && this.canvas.height === height)
             return;
```

Comment:
while you're here, maybe flip the logic and get rid of the return.

Reply:
Done!

Reply:
perfect.